diff --git a/render_omni_gui.py b/render_omni_gui.py
index 00f2bc9..b7480fc 100644
--- a/render_omni_gui.py
+++ b/render_omni_gui.py
@@ -432,13 +432,13 @@ class GaussianSplattingGUI:
 
     def pca(self, X, n_components=3):
         n = X.shape[0]
-        mean = torch.mean(X, dim=0)
+        mean = torch.mean(X, axis=0)
         X = X - mean
         covariance_matrix = (1 / n) * torch.matmul(X.T, X).float()  # An old torch bug: matmul float32->float16, 
-        eigenvalues, eigenvectors = torch.eig(covariance_matrix, eigenvectors=True)
-        eigenvalues = torch.norm(eigenvalues, dim=1)
-        idx = torch.argsort(-eigenvalues)
-        eigenvectors = eigenvectors[:, idx]
+        eigenvalues, eigenvectors = torch.linalg.eig(covariance_matrix)
+        # Ignore im. component
+        idx = torch.argsort(-eigenvalues.real)
+        eigenvectors = eigenvectors[:, idx].real
         proj_mat = eigenvectors[:, 0:n_components]
         
         return proj_mat
